### DISCUSSION TITLE
vi.mock does not work if I try to mock a module of another third-party module.

### DISCUSSION THREAD

**Discussion by JieamI** [vitest-dev/vitest#3295] [General]
suspect I have a test file `index.test.ts`, which imports a third-party module named 'A' from node_modules, and the module 'A' also imports a third-party module named 'B'.
when I try to mock module B, it does not take effect in module A.
```
// index.test.ts
vi.mock('B', () => ({}))
```
I'm not sure if this behavior is expected, or maybe there is something wrong with my configuration.

**Comment by sheremet-va** [comment]
Vitest doesn't process imports inside `node_modules`. If you want mocks to take effect, use `deps.inline` on the module that imports the one you want to mock (in your case it is the module A).
